From 2beea4afe7d163e53c1c0281b001fd4629f0aee5 Mon Sep 17 00:00:00 2001
From: Roman Stratiienko <r.stratiienko@gmail.com>
Date: Tue, 11 Apr 2023 23:31:52 +0300
Subject: [PATCH] GLODROID/SUNXI: FFMPEG: Enable hardware codecs

Signed-off-by: Roman Stratiienko <r.stratiienko@gmail.com>
---
 ffmpeg.mk | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/ffmpeg.mk b/ffmpeg.mk
index 9086afd..52b214d 100644
--- a/ffmpeg.mk
+++ b/ffmpeg.mk
@@ -41,6 +41,13 @@ FFMPEG_GEN_PKGCONFIGS += dav1d:$(DAV1D_VERSION)
 FFMPEG_DEFINITIONS += --enable-libdav1d
 endif
 
+ifneq ($(filter true, $(BOARD_FFMPEG_ENABLE_REQUEST_API)),)
+LOCAL_SHARED_LIBRARIES += libdrm libudev
+FFMPEG_GEN_PKGCONFIGS += libdrm libudev
+FFMPEG_DEFINITIONS += --enable-libdrm --enable-libudev --enable-v4l2-request
+LOCAL_C_INCLUDES := vendor/devices-community/pine64_pinephone/codecs/request_api_headers_v4
+endif
+
 #-------------------------------------------------------------------------------
 
 LOCAL_MULTILIB := first
-- 
2.39.2

