From 97904e01f000a0c93b51241378cdf9334f07091e Mon Sep 17 00:00:00 2001
From: Ondrej Jirman <megous@megous.com>
Date: Tue, 1 Sep 2020 17:45:06 +0200
Subject: [PATCH] usb: typec: anx7688: Disable current limit forcing for now

Signed-off-by: Ondrej Jirman <megous@megous.com>
---
 drivers/usb/typec/anx7688.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/usb/typec/anx7688.c b/drivers/usb/typec/anx7688.c
index 0a5c547c5949..82692a54c78b 100644
--- a/drivers/usb/typec/anx7688.c
+++ b/drivers/usb/typec/anx7688.c
@@ -1426,7 +1426,8 @@ static void anx7688_handle_vbus_in_notify(struct anx7688* anx7688)
 	struct device *dev = anx7688->dev;
 	int ret;
 
-	if (anx7688->current_limit) {
+	/* PD charger doesn't like this for some reason, so it's disabled for now */
+	if (anx7688->current_limit && false) {
 		ret = power_supply_get_property(anx7688->vbus_in_supply,
 						POWER_SUPPLY_PROP_INPUT_CURRENT_LIMIT,
 						&psy_val);
-- 
2.34.1

